src/core/runtime/preload-browser.js

```diff
@@ -1,27 +1,27 @@
 /* eslint-env browser */
 'use strict'
 
+const { default: PQueue } = require('p-queue')
+const { default: ky } = require('ky-universal')
 const debug = require('debug')
 
 const log = debug('ipfs:preload')
 log.error = debug('ipfs:preload:error')
 
+// browsers limit concurrent connections per host,
+// we don't want preload calls to exhaust the limit (~6)
+const _httpQueue = new PQueue({ concurrency: 4 })
+
 module.exports = function preload (url, callback) {
   log(url)
 
   const controller = new AbortController()
   const signal = controller.signal
 
-  fetch(url, { signal })
-    .then(res => {
-      if (!res.ok) {
-        log.error('failed to preload', url, res.status, res.statusText)
-        throw new Error(`failed to preload ${url}`)
-      }
-      return res.text()
-    })
-    .then(() => callback())
-    .catch(callback)
+  _httpQueue.add(async () => {
+    await ky.get(url, { signal })
+    setImmediate(callback)
+  }).catch((err) => setImmediate(() => callback(err)))
```

Comment:
No need to `setImmediate` already executed on a future tick. Passing callback as second parameter to `.then` is safer (instead of a catch) because it won't be called again if it throws.

```suggestion
  }).then(() => callback(), callback)
```

Reply:
This could just be:

```js
  _httpQueue.add(() => ky.get(url, { signal })).then(() => callback(), callback)
```

Reply:
@alanshaw  @hugomrdias raised concern that we need `setImmediate`, otherwise the callback will get caught inside the promise scope. Unless we are sure callback is "safe", i'd keep it

Reply:
yes the code inside the callback needs to break from the promise stack, because it handles stuff callback style. This is one of the reason we should always use packages to callbackify or promisify they handle this stuff.

Reply:
Yikes. I will now be double paranoid about this. 
Restored in https://github.com/ipfs/js-ipfs/pull/2304/commits/c9c1080ce8b8b94b833ce437753919ce79d8829d